Geometry/HGCalCommonData/plugins/DDHGCalTBModuleX.h

```diff
@@ -0,0 +1,69 @@
+#ifndef HGCalCommonData_DDHGCalTBModuleX_h
+#define HGCalCommonData_DDHGCalTBModuleX_h
+
+#include <map>
+#include <string>
+#include <vector>
+#include "DetectorDescription/Core/interface/DDTypes.h"
+#include "DetectorDescription/Core/interface/DDAlgorithm.h"
+#include <unordered_set>
+
+class DDHGCalTBModuleX : public DDAlgorithm {
+ 
+public:
+  //Constructor and Destructor
+  DDHGCalTBModuleX(); //
+  ~DDHGCalTBModuleX() override;
+  
+  void initialize(const DDNumericArguments & nArgs,
+                  const DDVectorArguments & vArgs,
+                  const DDMapArguments & mArgs,
+                  const DDStringArguments & sArgs,
+                  const DDStringVectorArguments & vsArgs) override;
+  void execute(DDCompactView& cpv) override;
+
+protected:
+
+  void          constructBlocks (const DDLogicalPart&, DDCompactView& cpv);
+  void          constructLayers (int block, int layerFront, int layerBack, 
+				 double zFront, double thick, bool ignore,
+				 const DDLogicalPart&, DDCompactView&);
+  void          positionSensitive(double zpos, int copyIn, int type, 
+				  double rmax, int ncrMax, bool ignoreCenter,
+				  const DDLogicalPart&, DDCompactView& cpv);
+
+private:
+
+  static constexpr double factor_ = 0.5*sqrt(2.0);
+  static constexpr double tolerance_ = 0.00001;
+  const double            tan30deg_;
```

Comment:
```suggestion
  static constexpr double tan30deg_ = tan(30._deg);
```

Reply:
Isn't `tan30deg_` a fixed constant like `factor`? Shouldn't it be `static`? You might have to include <cmath> in this header file for the tan function. Is that a problem?

Reply:
Moved